terraform/aws/modules/network/main.tf

```diff
@@ -162,17 +162,17 @@ resource "aws_lb" "km_lb" {
 resource "aws_lb_target_group" "km_lb_target" {
   name        = "km-lb-target-group-${var.environment}"
   port        = 80
-  protocol    = "HTTP"
+  protocol    = "HTTPS"
   vpc_id      = aws_vpc.km_vpc.id
   target_type = "ip"
-  depends_on = [ aws_lb.km_lb ]
+  depends_on  = [aws_lb.km_lb]
 }
 
 # Redirect all traffic from the ALB to the target group
 resource "aws_lb_listener" "km_frontend_listener" {
   load_balancer_arn = aws_lb.km_lb.arn
   port              = "80"
-  protocol          = "HTTP"
+  protocol          = "HTTPS"
```

Comment:
**Accurics remediation**
 
**[HIGH]: Ensure one HTTPS listener is configured for AWS Load balancer through AWS LB Listener**
 
**Vulnerability**
Not configuring AWS Load balancer to have one HTTPS listener could impact the confidentiality of data in transit.
 
**Remediation**
It is recommended to configure one HTTPS listener for AWS Load balancer. This can be achieved by declaring a resource 'aws_lb_listener' in your iac and make sure 'port' field is set to 443.

source: accurics-